src/plonk/lookup/prover.rs

```diff
@@ -306,11 +306,11 @@ impl<'a, C: CurveAffine> Permuted<'a, C> {
                 let mut right = z[prev_idx];
                 let mut input_term = self.unpermuted_input_columns
                     .iter()
-                    .fold(C::Scalar::zero(), |acc, input| acc * &theta + &input[i]);
+                    .fold(C::Scalar::zero(), |acc, input| acc * &*theta + &input[i]);
 
                 let mut table_term = self.unpermuted_table_columns
                     .iter()
-                    .fold(C::Scalar::zero(), |acc, table| acc * &theta + &table[i]);
+                    .fold(C::Scalar::zero(), |acc, table| acc * &*theta + &table[i]);
 
                 input_term += &(*beta);
                 table_term += &(*gamma);
```

Comment:
```suggestion
                input_term += &*beta;
                table_term += &*gamma;
```
and similarly elsewhere.

Reply:
We're likely to remove all this dereferencing later anyway, by implementing the necessary operator traits after @ebfull is done with refactoring.